src/App.tsx

```diff
@@ -1,61 +1,119 @@
+import { useState } from 'react';
+
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const todos = todosFromServer.map(todo => ({
+  ...todo,
+  user: usersFromServer.find(user => user.id === todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
+
+  const [newTodos, setNewTodos] = useState<Todo[]>(todos);
+
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const maxTodoId = newTodos.reduce(
+    (max, todo) => (todo.id < max ? max : todo.id),
+    0,
+  );
+
+  const handleTitleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    let newValue = e.target.value;
+
+    newValue = newValue.replace(/[^a-zA-Zа-яА-Я0-9\s]/g, '');
+    setTitle(newValue);
+    setTitleError(false);
+  };
+
+  const handleUserChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUserId(+e.target.value);
+    setUserError(false);
+  };
+
+  const handleTitleError = () => {
+    if (!title) {
+      setTitleError(true);
+    }
+  };
+
+  const handleUserError = () => {
+    if (!selectedUserId) {
+      setUserError(true);
+    }
+  };
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    handleTitleError();
```

Comment:
Looks like unnecessary complication here. If you want to extract this logic from the handleSubmit function (which is 100% a good idea) - will be enough to create a `validate` function. 

If you have 10 values to validate creating a separate function is just too much.